sdk/go/kms_test.go

```diff
@@ -0,0 +1,139 @@
+package ehsm
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"testing"
+
+	"github.com/tjfoc/gmsm/sm3"
+)
+
+func TestECSignVerify(t *testing.T) {
+	client, err := NewClient()
+	if err != nil {
+		t.Error(err)
+	}
+	keyspec := []string{"EH_EC_P224", "EH_EC_P256", "EH_EC_P256K", "EH_EC_P384", "EH_EC_P521"}
+	msg := "test for signature"
+	msg_digest256 := sha256.Sum256([]byte(msg))
+	message_type := []string{"EH_RAW", "EH_DIGEST"}
+	for _, keyspec := range keyspec {
+		keyid, err := client.CreateKey(keyspec, "EH_INTERNAL_KEY", "EH_KEYUSAGE_SIGN_VERIFY")
+		if err != nil {
+			t.Error(err)
+		}
+		for _, message_type := range message_type {
+			signature256, err := client.Sign(keyid, "EH_PAD_NONE", "EH_SHA_256", message_type, base64.StdEncoding.EncodeToString(msg_digest256[:]))
+			if err != nil {
+				t.Error(err)
+			}
+			result256, err := client.Verify(keyid, "EH_PAD_NONE", "EH_SHA_256", message_type, base64.StdEncoding.EncodeToString(msg_digest256[:]), signature256)
+			if result256 != true {
+				t.Error(err)
+			}
+		}
+
+	}
+}
+
+func TestRSASignVerifyt(t *testing.T) {
+	client, err := NewClient()
+	if err != nil {
+		t.Error(err)
+	}
+	keyspec := []string{"EH_RSA_3072", "EH_RSA_4096", "EH_RSA_2048"}
+	msg := "test for signature"
+	msg_digest256 := sha256.Sum256([]byte(msg))
+	message_type := []string{"EH_RAW", "EH_DIGEST"}
+	padding_mode := []string{"EH_RSA_PKCS1", "EH_RSA_PKCS1_PSS"}
+	for _, keyspec := range keyspec {
+		keyid, err := client.CreateKey(keyspec, "EH_INTERNAL_KEY", "EH_KEYUSAGE_SIGN_VERIFY")
+		if err != nil {
+			t.Error(err)
+		}
+		for _, message_type := range message_type {
+			for _, padding_mode := range padding_mode {
+				signature256, err := client.Sign(keyid, padding_mode, "EH_SHA_256", message_type, base64.StdEncoding.EncodeToString(msg_digest256[:]))
+				if err != nil {
+					t.Error(err)
+				}
+				result256, err := client.Verify(keyid, padding_mode, "EH_SHA_256", message_type, base64.StdEncoding.EncodeToString(msg_digest256[:]), signature256)
+				if result256 != true {
+					t.Error(err)
+				}
+			}
+		}
+	}
```

Comment:
for _, keyspec := range keyspec {
		keyid, err := client.CreateKey(keyspec, "EH_INTERNAL_KEY", "EH_KEYUSAGE_SIGN_VERIFY")
		if err != nil {
			t.Error(err)
		}
			for _, padding_mode := range padding_mode {
				signature_digest, err := client.Sign(keyid, padding_mode, "EH_SHA_256", “EH_DIGEST”, base64.StdEncoding.EncodeToString(msg_digest256[:]))
				if err != nil {
					t.Error(err)
				}
				result_digest, err := client.Verify(keyid, padding_mode, "EH_SHA_256", EH_DIGEST, base64.StdEncoding.EncodeToString(msg_digest256[:]), signature256)
				if result_digest!= true && err != nil {
					t.Error(err)
				}

                                signature_raw, err := client.Sign(keyid, padding_mode, "EH_SHA_256", “EH_RAW”, base64.StdEncoding.EncodeToString(msg[:]))
				if err != nil {
					t.Error(err)
				}
				result_raw, err := client.Verify(keyid, padding_mode, "EH_SHA_256", EH_DIGEST, base64.StdEncoding.EncodeToString(msg[:]), signature_raw)
				if result_raw!= true && err != nil {
					t.Error(err)
				}
			}
	}